Added sign up logic and changed to use MySQL DB

bananapizzza opened the pull request.

bananapizzza committed:
Added logic for signing up. (Logic for adding a new user to the JSON file.)

bananapizzza committed:
Changed structure of checking login info logic. (Made a new func named checkLoginInfo)

bananapizzza committed:
Added sign up checking logic.

bananapizzza committed:
Changed the data management resource for users from file to MySQL DB.
Changed the logic for signing up to use MySQL DB.

bananapizzza committed:
Changed the logic for login to use MySQL DB.
Deleted user_list.json.

xaveng commented on package.json:
```diff
@@ -12,6 +12,7 @@
     "cors": "^2.8.5",
     "express": "^4.17.1",
-    "jquery": "^3.4.1"
+    "jquery": "^3.4.1",
```
Why do we need jquery?

xaveng replied:
npm uninstall jquery

bananapizzza replied:
done

xaveng commented on src/backendLogic.js:
```diff
@@ -5,23 +5,30 @@ const path = require('path');
 const readFileAsync = util.promisify(fs.readFile);
+const writeFileAsync = util.promisify(fs.writeFile);
 const rootPath = path.parse(appRoot).dir;
+ const queryDBAsync = util.promisify(pool.query).bind(pool);
```
minor: indentation

xaveng commented on src/backendLogic.js:
```diff
@@ -5,23 +5,30 @@ const path = require('path');
+    database: "random_workout"
+});
+
 const readFileAsync = util.promisify(fs.readFile);
```
Do we need this ?

bananapizzza replied:
deleted

xaveng commented on src/backendLogic.js:
```diff
@@ -38,26 +45,102 @@ app.get('/', async (req, res) => {
+            res.send(JSON.stringify(LOGIN_SUCCESS));
+            break;
+        case WRONG_PASSWORD:
+            res.statusCode = 401;
```
Let's use this with http code name, not just numbers.
https://www.npmjs.com/package/http-status-codes

bananapizzza replied:
okie

xaveng commented on src/backendLogic.js:
```diff
@@ -38,26 +45,102 @@ app.get('/', async (req, res) => {
+            if (queryResult.length > 0) {
+                return ALREADY_USED_ID;
+            } else {
+                return SIGN_UP_SUCCESS;
```
Looks like we are checking if this user is in the DB or not. Let's make a separate keyword for this.

bananapizzza replied:
done

xaveng commented on src/backendLogic.js:
```diff
@@ -38,26 +45,102 @@ app.get('/', async (req, res) => {
+            await queryDBAsync(sql, values);
+            console.log("Added new user to DB");
+        } catch (err) {
+            console.log("addNewUserToDB err: "+err);
```
minor: indentation.

xaveng reviewed (commented):
As this backendLogic is getting larger, we could think of separating this file into each files. E.g) loginProcessor.js something like this. :)

bananapizzza committed:
Changed http status code to use 'http-status-code' module.
Added AVAILABLE_ID status for checking sign up info.
Uninstalled unused jquery module.
